node/src/accountant/mod.rs

```diff
@@ -4234,26 +4172,131 @@ mod tests {
             unban_below_gwei: 0,
         };
-        let higher_corner_timestamp = payment_thresholds.maturity_threshold_sec;
+        //the very end on the left is forbid so we have to estimate
+        let higher_corner_timestamp = payment_thresholds.maturity_threshold_sec + 1;
         let middle_point_timestamp = payment_thresholds.maturity_threshold_sec
             + payment_thresholds.threshold_interval_sec / 2;
         let lower_corner_timestamp =
             payment_thresholds.maturity_threshold_sec + payment_thresholds.threshold_interval_sec;
         let tested_fn = |payment_thresholds: &PaymentThresholds, time| {
-            PayableThresholdToolsReal {}.calculate_payout_threshold(payment_thresholds, time)
-                as i128
+            PayableThresholdsGaugeReal {}
+                .calculate_payout_threshold_in_gwei(payment_thresholds, time) as i128
                 * WEIS_OF_GWEI
         };
 
-        assert_on_sloped_segment_of_payment_thresholds_and_its_proper_alignment(
-            tested_fn,
-            payment_thresholds,
-            higher_corner_timestamp,
-            middle_point_timestamp,
-            lower_corner_timestamp,
+        let higher_corner_point = tested_fn(&payment_thresholds, higher_corner_timestamp);
+        let middle_point = tested_fn(&payment_thresholds, middle_point_timestamp);
+        let lower_corner_point = tested_fn(&payment_thresholds, lower_corner_timestamp);
+
+        let allowed_imprecision = 1 * WEIS_OF_GWEI;
+        let ideal_template_higher = payment_thresholds.debt_threshold_gwei as i128 * WEIS_OF_GWEI;
+        let ideal_template_middle = ((payment_thresholds.debt_threshold_gwei
+            - payment_thresholds.permanent_debt_allowed_gwei)
+            / 2
+            + payment_thresholds.permanent_debt_allowed_gwei)
+            as i128
+            * WEIS_OF_GWEI;
+        let ideal_template_lower =
+            payment_thresholds.permanent_debt_allowed_gwei as i128 * WEIS_OF_GWEI;
+        assert!(
+            higher_corner_point <= ideal_template_higher + allowed_imprecision
+                && ideal_template_higher - allowed_imprecision <= higher_corner_point,
+            "ideal: {}, real: {}",
+            ideal_template_higher,
+            higher_corner_point
+        );
+        assert!(
+            middle_point <= ideal_template_middle + allowed_imprecision
+                && ideal_template_middle - allowed_imprecision <= middle_point,
+            "ideal: {}, real: {}",
+            ideal_template_middle,
+            middle_point
+        );
+        assert!(
+            lower_corner_point <= ideal_template_lower + allowed_imprecision
+                && ideal_template_lower - allowed_imprecision <= lower_corner_point,
+            "ideal: {}, real: {}",
+            ideal_template_lower,
+            lower_corner_point
         )
     }
 
+    #[test]
+    fn testing_granularity_calculate_sloped_threshold_by_time() {
```

Comment:
It would be nice to run the gap_tester on the real PaymentThresholds, just so that if somebody modified them to be stupid, we'd know immediately.